arviz/data/inference_data.py

```diff
@@ -423,11 +430,13 @@ def to_netcdf(
                 kwargs = {}
                 if compress:
                     kwargs["encoding"] = {var_name: {"zlib": True} for var_name in data.variables}
+                if HAS_H5NETCDF:
+                    kwargs['engine'] = 'h5netcdf'
                 data.to_netcdf(filename, mode=mode, group=group, **kwargs)
                 data.close()
                 mode = "a"
         else:  # creates a netcdf file for an empty InferenceData object.
-            empty_netcdf_file = nc.Dataset(filename, mode="w", format="NETCDF4")
+            empty_netcdf_file = nc.Dataset(filename, mode="w")
```

Comment:
Is the format always netcdf4 with h4netcdf?

Reply:
Hi @OriolAbril, 

The basic reason for this is to remove the dependency on the netCDF4 C library. Since netCDF4 is a layer on top of HDF5, you could imagine using a basic HDF5 library in python to implement the netCDF4 standard, instead of calling netCDF4 C routines. `h5netcdf` does this using the `h5py` library. 

I find installing the netCDF4 C binaries to be quite painful, especially when I need to support linux, windows, intel OSX, Silicon OSX, etc. On Windows, there are some times issues with binary incompatibilities between the HDF5 libraries installed by `h5py` and `netCDF4` on `conda` Switching to using `h5py` as the backend would make the install significantly more straightforward. I think most scientific python users probably already have `h5py` installed, so practically I would imagine this wouldn't actually be a new dependency in those cases. 

I see where you're coming from with something called `legacy` showing up in an import - but I believe that the legacy API you saw allows `h5netcdf` to be used as a 'drop-in' replacement for `netCDF4` while the 'new' API looks more like `h5py`, rather than this being some kind of deprecated support issue.

> Is the format always netcdf4 with h4netcdf? 

No idea.

Reply:
Also, FWIW `xarray` seems to fully support `h5netcdf` as a backend. That's an appeal to authority, but it does seem to be ok for them!